CONTRIBUTING.md

```diff
@@ -0,0 +1,123 @@
+## For maintainers: Reviewing a PR
+
+The formal code reviews are done on GitHub.  Reviewers are to look for all of the usual things:
+
+
+
+*   Coding style follows the [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html)
+*   Identify potential functional problems.
+*   Identify code duplication.
+*   Ensure the unit tests have enough coverage.
+
+When looking for functional problems, there are some common problems reviewers should pay particular attention to:
+
+
+
+*   Does the code work for both Shader (Vulkan and OpenGL) and Kernel (OpenCL) scenarios?  The respective SPIR-V dialects are slightly different.
+*   Changes are made to a container while iterating through it.  You have to be careful that iterators are not invalidated or that elements are not skipped.
+*   C++11 and VS2013.  We generally assume that we have a C++11 compliant compiler.  However, on Windows, we still support Visual Studio 2013, which is not fully C++11 compliant. See [here](https://msdn.microsoft.com/en-us/library/hh567368.aspx).  In particular, note that it does not provide default move-constructors or move-assignments for classes.  In general, r-value references do not work the way you might assume they do.
+*   For SPIR-V transforms: The module is changed, but the analyses are not updated.  For example, a new instruction is added, but the def-use manager is not updated.  Later on, it is possible that the def-use manager will be used, and give wrong results.
+
+
+## For maintainers: Merging a PR
+
+We intend to maintain a linear history on the GitHub master branch, and the build and its tests should pass at each commit in that history.  A linear always-working history is easier to understand and to bisect in case we want to find which commit introduced a bug.
+
+These are the steps to follow to ensure this happens:
+
+
+
+*   \<userid\> is the user id of the author of the PR.
+*   \<url\> is the url to that user's fork of SPIRV-Tools.  Will often look like 
+
+    ```
+    git@github.com:<userid>/SPIRV-Tools.git
+    ```
+
+
+*   \<branch\> is the name of the branch that contains the PR
+*   It is assumed that upstream points to [git@github.com](mailto:git@github.com):KhronosGroup/SPIRV-Tools.git.  You can check this by doing
+
+    ```
+    git remote -v
+    ```
+
+*   Make sure your local copy of the master branch is up to date:
+
+    ```
+    git pull
+    ```
+
+*   Get a local copy of the branch containing the PR.
+  
+    ```
+    git remote add <userid> <url>
+    git fetch <userid> <branch>
+    git checkout <userid>/<branch>
+    git branch -D <branch> # if it already exists
+    git checkout -b <branch>  # assuming it does not collide.
+    ```
+  
+*   Rebase the branch on top of the master branch.  If there are conflicts, send it back to the author and ask them to rebase.  During the interactive rebase be sure to squash all of the commits down to a single commit.
+
+    ```
+    git branch -u master
+    git rebase -i
+    ```
```

Comment:
I intend to contribute my script that does all these steps.  I'll add it as an alternative later, or we could simply point folks to it.

Reply:
We can create a reference to your script when it if available.  For now, I'll leave it as is.

Reply:
SGTM